client/my-sites/site-settings/test/form-general.jsx

```diff
@@ -88,4 +131,84 @@ describe( 'SiteSettingsFormGeneral ', () => {
 			expect( comp.find( 'Banner' ).length ).toBe( 0 );
 		} );
 	} );
+
+	describe( 'Privacy picker - simple sites', () => {
+		let testProps;
+
+		beforeEach( () => {
+			testProps = {
+				...props,
+				siteIsJetpack: false,
+				site: { plan: PLAN_PERSONAL },
+				fields: {
+					blog_public: 1,
+					wpcom_coming_soon: 0,
+				},
+				withComingSoonOption: true,
+				updateFields: fields => {
+					testProps.fields = fields;
+				},
+			};
+		} );
+
+		test( `Should have 4 visibility options`, () => {
+			const { container } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+			expect( container.querySelectorAll( '[name="blog_public"]' ).length ).toBe( 4 );
+		} );
+
+		[
+			[ 'Coming soon', '.is-coming-soon [name="blog_public"]', 1 ],
+			[ 'Public', '.is-public [name="blog_public"]', -1 ],
+			[ 'Hidden', '.is-hidden [name="blog_public"]', -1 ],
+			[ 'Private', '.is-private [name="blog_public"]', 1 ],
+		].forEach( ( [ name, selector, initialBlogPublic ] ) => {
+			test( `${ name } option should be selectable`, () => {
+				testProps.fields.blog_public = initialBlogPublic;
+				const { container, rerender } = render( <SiteSettingsFormGeneral { ...testProps } /> );
+
+				const radioButton = container.querySelector( selector );
```

Comment:
In testing-library it's more idiomatic to [select form control by using their label text](https://testing-library.com/docs/guide-which-query). That way you're 1) testing from a users perspective 2) testing a11y for free 🎉. Querying by selector is sort of a last resort. 

If you destructure the `getByLabelText` function from the result of the `render` function then you can use it like this:

```suggestion
				const radioButton = getByLabelText( name );
```

You'd also need to change `'Coming soon'` to `'Coming Soon'` and `'Hidden'` to `'Do not allow search engines to index my site'` above.